client/blocks/login/index.jsx

```diff
@@ -137,13 +137,19 @@ class Login extends Component {
 		if ( this.props.onTwoFactorRequested ) {
 			this.props.onTwoFactorRequested( authType );
 		} else {
+			const currentURLQueryParameters = Object.fromEntries(
+				new URL( this.props.redirectTo ).searchParams.entries()
+			);
 			page(
 				login( {
 					isJetpack: this.props.isJetpack,
 					isGutenboarding: this.props.isGutenboarding,
 					// If no notification is sent, the user is using the authenticator for 2FA by default
 					twoFactorAuthType: authType,
 					locale: this.props.locale,
+					redirectTo: isPartnerSignupQuery( currentURLQueryParameters )
+						? this.props.redirectTo
+						: false,
```

Comment:
Is this redirectTo used to redirect a user in this scenario? (or is it just used in another isPartnerSignupQuery call?)

If not, we can instead just change isPartnerSignupQuery to detect a new named param like isPartnerSignup or something and be more specific in passing it through.

This will be useful for the create account step too I think as it has a similar problem.